Extension/src/LanguageServer/configurations.ts

```diff
@@ -2029,7 +2029,13 @@ export class CppProperties {
             // and extend that pattern to the next quote before and next quote after it.
             const pattern: RegExp = new RegExp(`"[^"]*?(?<="|;)${escapedPath}(?="|;).*?"`, "g");
             const configMatches: string[] | null = curText.match(pattern);
-            if (configMatches && !escapedPath.includes('*')) {
+
+            let globPath: boolean = false;
+            if (curPath.includes("*") && !curPath.endsWith("*")) {
```

Comment:
Actually, it needs to be able handle paths with glob patterns in the middle and "*" at the end, in which case it shouldn't squiggle (the current code does still), i.e. the `includes(*)` should only be counted as glob pattern if the index is not at the end, i.e. maybe switch to calling some indexOf function (or whatever it's called) and checking if it's not equal to the last character.